File: fatorial.hs
import Text.Printf

fatorial :: Integer -> Integer
fatorial 0 = 1
fatorial n = n * fatorial (n-1)

main :: IO ()
main = do
    num' <- getLine
    let num = read num' :: Integer
    let fat = fatorial num
    
    putStrLn $ printf "%d" fat

File: fatorial.ex
defmodule Fatorial do
  def run do
    IO.read(:stdio, :line)
    |> String.trim
    |> String.to_integer
    |> fatorial
    |> IO.puts
  end

  def fatorial(0), do: 1
  def fatorial(num), do: num * fatorial(num - 1)
end
Fatorial.run
